src/rulesets/Base/mapreduce.jl

```diff
@@ -10,9 +10,10 @@ function rrule(::typeof(sum), x::AbstractArray{T}; dims=:) where {T<:Number}
     y = sum(x; dims=dims)
     function sum_pullback(ȳ)
         # broadcasting the two works out the size no-matter `dims`
-        x̄ = broadcast(x, ȳ) do xi, ȳi
-            ȳi
-        end
+        x̄ = InplaceableThunk(
+            @thunk(broadcast((_,y1)->y1, x, ȳ)), # last∘tuple
+            x -> x .+= x̄
```

Comment:
this line is wrong isn't it?
Shouldn't it be:
```suggestion
            x -> x .+= ȳ
```

Reply:
Good catch. 

How disturbed should I be that tests didn't fail here?

Reply:
Moderately. We should work out why that is happenning.